pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,326 @@
+	if openstackclient.IgnoreNotFoundError(err) != nil {
+		return err
+	}
+
+	if len(instancenstatus) != 0 {
+		return err
+	}
+
+	// check if the instance already exists and has an IP
+	endpoints, err := getInstanceEndpoints(&instancenstatus[0], opt)
+	if err != nil {
+		return err
+	}
+
+	if !endpoints.Ready() {
+		return &ctrlerror.RequeueAfterError{
+			// requeue rather soon, so that the user (most likely gardenctl eventually)
+			// doesn't have to wait too long for the public endpoint to become available
+			RequeueAfter: 5 * time.Second,
+			Cause:        fmt.Errorf("bastion instance has no public/private endpoints yet"),
+		}
+	}
+
+	// once a public endpoint is available, publish the endpoint on the
+	// Bastion resource to notify upstream about the ready instance
+	return controller.TryUpdateStatus(ctx, retry.DefaultBackoff, a.client, bastion, func() error {
+		bastion.Status.Ingress = *endpoints.public
+		return nil
+	})
+}
+
+func ensurePublicIPAddress(opt *Options, openstackClientFactory openstackclient.Factory) (*floatingips.FloatingIP, error) {
+	info, err := getFipbyName(openstackClientFactory, opt.BastionInstanceName)
+	if err != nil {
+		return nil, err
+	}
+
+	if info != nil && info[0].Status == "ACTIVE" {
+		return &info[0], nil
+	}
+
+	logger.Info("creating new bastion Public IP")
+
+	externalFipInfo, err := getExternalNetworkInfoByName(openstackClientFactory, opt.FloatingPoolName)
+	if err != nil {
+		return nil, err
+	}
+
+	createOpts := floatingips.CreateOpts{
+		FloatingNetworkID: externalFipInfo[0].ID,
+		Description:       opt.BastionInstanceName,
+	}
+
+	fip, err := createFloatingIP(openstackClientFactory, createOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get (create) public ip address: %w", err)
+	}
+
+	return fip, nil
+}
+
+func ensureComputeInstance(logger logr.Logger, openstackClientFactory openstackclient.Factory, opt *Options) (*servers.Server, error) {
+	getInstance, err := getBastionInstance(openstackClientFactory, opt.BastionInstanceName)
+	if openstackclient.IgnoreNotFoundError(err) != nil {
+		return nil, err
+	}
+
+	if len(getInstance) != 0 {
+		return &getInstance[0], err
+	}
+
+	logger.Info("Creating new bastion compute instance")
+
+	networkingClient, err := openstackClientFactory.Networking()
+	if err != nil {
+		return nil, err
+	}
+
+	networkInfo, err := networkingClient.GetNetworkByName(opt.ShootName)
+	if err != nil {
+		return nil, err
+	}
+
+	createOpts := servers.CreateOpts{
+		Name:           opt.BastionInstanceName,
+		FlavorRef:      opt.FlavorRef,
+		ImageRef:       opt.ImageRef,
+		SecurityGroups: []string{opt.SecurityGroup},
+		Networks:       []servers.Network{{UUID: networkInfo[0].ID}},
+		UserData:       opt.UserData,
+	}
+
+	instance, err := createBastionInstance(openstackClientFactory, createOpts)
+	if err != nil {
+		return nil, fmt.Errorf("%w, failed to create bastion compute instance", err)
+	}
+
+	if instance != nil {
+		return instance, err
+	}
```

Comment:
I guess due to CCloud being unstable or network issues sometimes, then I add this part. but I think you are right we can get rid of this check. because there are another check for the instance and public status to associate the public IP with the instance later. 

![image](https://user-images.githubusercontent.com/42234376/149445538-8191a89d-8e45-4bf0-98b6-5a3f98554b24.png)